pac4j-saml/src/main/java/org/pac4j/saml/client/SAML2ClientConfiguration.java

```diff
@@ -15,285 +15,155 @@
  *
  * @author Misagh Moayyed
  * @since 1.7
  */
-public final class SAML2ClientConfiguration implements Cloneable {
+public final class SAML2ClientConfiguration extends AbstractSAML2ClientConfiguration {
 
     private String keystorePath;
 
-    private String keystorePassword;
-
-    private String privateKeyPassword;
-
     private String identityProviderMetadataPath;
 
-    private String identityProviderEntityId;
-
-    private String serviceProviderEntityId;
-
-    private int maximumAuthenticationLifetime;
-
-    private boolean forceAuth = false;
-
-    private String comparisonType = null;
-
-    private String destinationBindingType = SAMLConstants.SAML2_POST_BINDING_URI;
-
-    private String authnContextClassRef = null;
-
-    private String nameIdPolicyFormat = null;
-
-    private String serviceProviderMetadataPath;
-
-    private boolean forceServiceProviderMetadataGeneration;
-
-    private SAMLMessageStorageFactory samlMessageStorageFactory = new EmptyStorageFactory();
 
-    public SAML2ClientConfiguration() {}
-
-
-    private Collection<String> blackListedSignatureSigningAlgorithms;
-    private List<String> signatureAlgorithms;
-    private List<String> signatureReferenceDigestMethods;
-    private String signatureCanonicalizationAlgorithm;
-
-    public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath) {
-        this(keystorePath, keystorePassword, privateKeyPassword, identityProviderMetadataPath, null, null);
+    // ------------------------------------------------------------------------------------------------------------------------------------
+    
+    
+    public SAML2ClientConfiguration() {
+    	super();
     }
 
     public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath,
-                                    final String identityProviderEntityId, final String serviceProviderEntityId) {
-        this.keystorePath = keystorePath;
-        this.keystorePassword = keystorePassword;
-        this.privateKeyPassword = privateKeyPassword;
-        this.identityProviderMetadataPath = identityProviderMetadataPath;
-        this.identityProviderEntityId = identityProviderEntityId;
-        this.serviceProviderEntityId = serviceProviderEntityId;
-
-        CommonHelper.assertNotBlank("keystorePath", this.keystorePath);
-        CommonHelper.assertNotBlank("keystorePassword", this.keystorePassword);
-        CommonHelper.assertNotBlank("privateKeyPassword", this.privateKeyPassword);
-        CommonHelper.assertNotBlank("identityProviderMetadataPath", this.identityProviderMetadataPath);
-
-        final BasicSignatureSigningConfiguration config = DefaultSecurityConfigurationBootstrap.buildDefaultSignatureSigningConfiguration();
```

Comment:
Only the last part - BasicSignatureSigningConfiguration and 5 related attributes loaded from it - are in the abstract configuration. The remaining 6 attributes stayed here. The deletion of the 6 assignments was a mistake - they are already back, together with the assertions. I am sorry for that.